### ISSUE TITLE
redis run as windows server crashed with some errors 

### ISSUE THREAD

**Issue by moonsound** [microsoftarchive/redis#229]
redis run as windows server crashed with some errors , when i start server , minutes late, Redis server crash, here are the errors.

=== REDIS BUG REPORT START: Cut & paste starting from here ===
[7220] 17 Mar 13:59:48.075 # === ASSERTION FAILED ===
[7220] 17 Mar 13:59:48.075 # ==> ..\src\ziplist.c:405 'NULL' is not true

=== REDIS BUG REPORT START: Cut & paste starting from here ===
[3672] 17 Mar 14:06:12.097 # ------------------------------------------------
[3672] 17 Mar 14:06:12.098 # !!! Software Failure. Press left mouse button to continue
[3672] 17 Mar 14:06:12.099 # Guru Meditation: "decrRefCount against refcount <= 0" #..\src\object.c:251
[3672] 17 Mar 14:06:12.100 # ------------------------------------------------

sometimes reidis exit without erros。

i user c# client  StackExchange.Redis as driver, client update stock price, vol witch store in redis server high frequence about 2700 times in 30 seconds.

for example：

```
                string volkey = string.Format("vol_{0}_{1}", ndate, stockid);
                string openkey = string.Format("open_{0}_{1}", ndate, stockid);
                string highkey = string.Format("high_{0}_{1}", ndate, stockid);
                string lowkey = string.Format("low_{0}_{1}", ndate, stockid);
                string yesterdaypricekey = string.Format("yesterdayprice_{0}_{1}", ndate, stockid);

                var tran = db.CreateTransaction();

                tran.StringSetAsync(lastpricekey, daydata.nClosePrice, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);
                tran.StringSetAsync(volkey, daydata.Vol, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);
                tran.StringSetAsync(openkey, daydata.nOpenPrice, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);
                tran.StringSetAsync(highkey, daydata.nHighestPrice, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);
                tran.StringSetAsync(lowkey, daydata.nLowestPrice, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);
                tran.StringSetAsync(yesterdaypricekey, daydata.nYesterdayPrice, TimeSpan.FromDays(15), When.Always, CommandFlags.FireAndForget);

                tran.Execute();
```

**Comment by chester89** [comment]
Can you reduce the font size please?

**Comment by orangemocha** [comment]
Thanks for reporting this issue, @moonsound . Are you able to reproduce it consistently? If so, could you share more details about what the client is doing or, even better, a simplified script that I could use to reproduce the issue?

**Comment by moonsound** [comment]
already remove the configure file with large font， and add the code example for reproduce the issue。

**Comment by enricogior** [comment]
@moonsound, I am trying to reproduce the issue, using the code you provided, without success. Can you please provide some more info regarding your environment?
1. redis version
2. redis.conf if you are not using the default configuration
3. Windows version
4. hardware details (single/multi CPU, RAM, storage etc.)
Thank you.

**Comment by ricekeg** [comment]
I also have this problem,

```
            _._                                                  
       _.-``__ ''-._                                             
  _.-``    `.  `_.  ''-._           Redis 2.8.19 (00000000/0) 64 bit
```

  .-`.-``.``\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in stand alone mode
 |`-._`-...-` __...-.`-._|'`_.-'|     Port: 6379
 |`-._   `._    /     _.-'    |     PID: 1280
`-._    `-._`-./  _.-'    _.-'  
 |`-._`-._`-.**.-'    _.-'_.-'|  
 |    `-._`-._        _.-'_.-'    |           http://redis.io  
  `-._`-._`-.**.-'_.-'    _.-'  
 |`-._`-._`-.**.-'    _.-'_.-'|  
 |    `-._`-._        _.-'_.-'    |  
  `-._`-._`-.**.-'_.-'    _.-'  
      `-._    `-.__.-'    _.-'                                       
`-._        _.-'  
              `-.__.-'                                               

[1280] 29 May 09:24:40.861 # Server started, Redis version 2.8.19
[1280] 29 May 09:24:48.396 # 

=== REDIS BUG REPORT START: Cut & paste starting from here ===
[1280] 29 May 09:24:48.396 # === ASSERTION FAILED ===
[1280] 29 May 09:24:48.396 # ==> ..\src\ziplist.c:405 'NULL' is not true

My Windows version is Win2008 Server R2 Enterprise

**Comment by ricekeg** [comment]
The problem is resolved, reCreate a dump.rdb file, the old one was damaged

**Comment by orangemocha** [comment]
@moonsound this was most likely due to a memory corruption issue, which we have fixed in 2.8.21. Please upgrade and let us know if you are still having problems. Thanks!
